src/directory_schema/directory_schema.py

```diff
@@ -0,0 +1,111 @@
+import os
+
+from jsonschema import Draft7Validator, validate, ValidationError
+from yaml import dump as dump_yaml
+
+
+def to_dir_listing(dir_as_list, indent=''):
+    next_indent = indent + '    '
+    return ''.join([
+        '\n' + indent + item['name']
+        + to_dir_listing(
+            item['contents'] if 'contents' in item else [],
+            next_indent
+        )
+        for item in dir_as_list
+    ])
+
+
+def dir_to_dict(path):
+    '''
+    Walk the directory at `path`, and return a dict like that from `tree -J`:
+
+    [
+      {
+        "type": "directory",
+        "name": "some-directory",
+        "contents": [
+          { "type": "file", "name": "some-file.txt" }
+        ]
+      }
+    ]
+    '''
+
+    items_to_return = []
+    with os.scandir(path) as scan:
+        for entry in sorted(scan, key=lambda entry: entry.name):
+            is_dir = entry.is_dir()
+            item = {
+                'type': 'directory' if is_dir else 'file',
```

Comment:
Is there any way to codify this binary?  What else could be in a directory, if anything? I see you use an `enum` in the examples so I wonder if we can replicate that here somehow.

Reply:
I thought about having a schema to check that the data structure we produce is valid, in which case we would have something like
```
property:
  type:
    enum:
      - file
      - directory
```

But since we're creating the directory listing ourselves, and not taking input from elsewhere, I didn't think that was needed.